tests_e2e/tests/lib/virtual_machine_client.py

```diff
@@ -114,6 +117,23 @@ def restart(self, timeout: int = AzureClient._DEFAULT_TIMEOUT) -> None:
             operation_name=f"Restart {self._identifier}",
             timeout=timeout)
 
+        if not wait_for_boot:
+            return
+
+        start = datetime.datetime.now()
+        while datetime.datetime.now() < start + boot_timeout:
+            log.info("Waiting for VM %s to boot", self._identifier)
+            time.sleep(15)  # Note that we always sleep at least 1 time, to give the reboot time to start
+            instance_view = self.get_instance_view()
+            power_state = [s.code for s in instance_view.statuses if "PowerState" in s.code]
+            if len(power_state) != 1:
+                raise Exception(f"Could not find PowerState in the instance view statuses:\n{json.dumps(instance_view.statuses)}")
+            log.info("VM's Power State: %s", power_state[0])
+            if power_state[0] == "PowerState/running":
```

Comment:
how do we guarantee that this state is after vm restart? It's possible that vm request not performed due to something else but vm instance view is from previous state and this check could pass. I think checking timestamp of the status make sure this state is after vm restart?

Reply:
there is no guarantee, but we are not interested on the reboot itself, but rather on whether FIPS is enabled or not (which we check after this). if that checks fails because the machine did not reboot (or any other reason) then the test will fail

during my experiments, the reboot started within 2 or 3 seconds so 15 seconds wait seemed reasonable, but you are right that there is no guarantee about that

checking on some sort of timestamp is a good suggestion, though, and i'll add an extra check around that. the timestamp in the instance view is not appropriate for this check, though. i'll experiment with the service start in the agent log or the uptime of the system. thanks for the suggestion.

Reply:
done